bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/handler/MeteoAlerteHandler.java

```diff
@@ -0,0 +1,215 @@
+import org.openhab.binding.meteoalerte.internal.json.ApiResponse;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializer;
+
+/**
+ * The {@link MeteoAlerteHandler} is responsible for updating channels
+ * and querying the API
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class MeteoAlerteHandler extends BaseThingHandler {
+    private static final String URL = "https://public.opendatasoft.com/api/records/1.0/search/?dataset=risques-meteorologiques-copy&"
+            + "facet=etat_vent&facet=etat_pluie_inondation&facet=etat_orage&facet=etat_inondation&facet=etat_neige&facet=etat_canicule&"
+            + "facet=etat_grand_froid&facet=etat_avalanches&refine.nom_dept=";
+    private static final int TIMEOUT_MS = 30000;
+    private static final List<String> ALERT_LEVELS = Arrays.asList("Vert", "Jaune", "Orange", "Rouge");
+    private final Logger logger = LoggerFactory.getLogger(MeteoAlerteHandler.class);
+    private final Gson gson = new GsonBuilder()
+            .registerTypeAdapter(ZonedDateTime.class, (JsonDeserializer<ZonedDateTime>) (json, type,
+                    jsonDeserializationContext) -> ZonedDateTime.parse(json.getAsJsonPrimitive().getAsString()))
+            .create();
+
+    // Time zone provider representing time zone configured in openHAB configuration
+    private final TimeZoneProvider timeZoneProvider;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private String queryUrl = "";
+
+    public MeteoAlerteHandler(Thing thing, TimeZoneProvider timeZoneProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Météo Alerte handler.");
+
+        MeteoAlerteConfiguration config = getConfigAs(MeteoAlerteConfiguration.class);
+        logger.debug("config department = {}", config.department);
+        logger.debug("config refresh = {}", config.refresh);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        queryUrl = URL + config.department;
+        refreshJob = scheduler.scheduleWithFixedDelay(this::updateAndPublish, 0, config.refresh, TimeUnit.HOURS);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Météo Alerte handler.");
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateAndPublish();
+        } else {
+            logger.debug("The Meteo Alerte binding is read-only and can not handle command {}", command);
+        }
```

Comment:
```suggestion
        } 
```

Other bindings that support commands might be linked to these channels. Ignoring that silently is ok,